pkg/operator/ceph/csi/spec.go

```diff
@@ -601,6 +611,12 @@ func createCSIDriverInfo(ctx context.Context, clientset kubernetes.Interface, na
 	}
 
 	if apierrors.IsNotFound(err) {
+		if fsGroupPolicy != "" {
+			policy := k8scsi.FSGroupPolicy(fsGroupPolicy)
+			// FSGroupPolicy field is immutable, should be set only during
```

Comment:
> @travisn we do have two options here. Either the user can delete the CSI driver object and restart the operator pod. or Rook itself can detect the changed configuration and delete the CSI driver object and recreate it. which one do you prefer?

I expect it will be rare enough to change the default that we can just document the need to delete it and restart the operator. Or if deleting it when needed is almost as simple as writing the doc, maybe we should just do it. I'm fine either way, you can decide.

Reply:
Added the default now `ReadWriteOnceWithFSType` as per https://kubernetes-csi.github.io/docs/support-fsgroup.html and added logic to recreate the csidriver object if the policy is different to avoid manual intervention.